### ISSUE TITLE
Wrong type for version in ListACLs

### ISSUE THREAD

**Issue by davidglvn** [fastly/fastly-py#87]
**Version**
`1.1.0`

**What happened**
Running sample code for `list_acls` fails as the returned version in API request is string and the python expects for integer:
```
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/model_utils.py", line 1784, in get_allof_instances
    allof_instance = allof_class._from_openapi_data(**model_args, **constant_args)
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/model_utils.py", line 45, in wrapped_init
    return fn(_self, *args, **kwargs)
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/model/service_id_and_version.py", line 175, in _from_openapi_data
    setattr(self, var_name, var_value)
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/model_utils.py", line 184, in __setattr__
    self[attr] = value
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/model_utils.py", line 509, in __setitem__
    self.set_attribute(name, value)
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/model_utils.py", line 156, in set_attribute
    value = validate_and_convert_types(
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/model_utils.py", line 1569, in validate_and_convert_types
    converted_instance = attempt_convert_item(
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/model_utils.py", line 1448, in attempt_convert_item
    raise get_type_error(input_value, path_to_item, valid_classes,
fastly.exceptions.ApiTypeError: Invalid type for variable 'version'. Required value type is int and passed type was str at ['received_data'][0]['version']

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/davidg/Code/devops/lambdas/fastly/main.py", line 106, in <module>
    lambda_handler({}, "")
  File "/Users/davidg/Code/devops/lambdas/fastly/main.py", line 87, in lambda_handler
    fastly_get_service_acl(service_id, version_id)
  File "/Users/davidg/Code/devops/lambdas/fastly/main.py", line 35, in fastly_get_service_acl
    api_response = api_instance.list_acls(service_id, 15)
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/api/acl_api.py", line 690, in list_acls
    return self.list_acls_endpoint.call_with_http_info(**kwargs)
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/api_client.py", line 862, in call_with_http_info
    return self.api_client.call_api(
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/api_client.py", line 421, in call_api
    return self.__call_api(resource_path, method,
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/api_client.py", line 232, in __call_api
    return_data = self.deserialize(
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/api_client.py", line 337, in deserialize
    deserialized_data = validate_and_convert_types(
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/model_utils.py", line 1619, in validate_and_convert_types
    input_value[index] = validate_and_convert_types(
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/model_utils.py", line 1569, in validate_and_convert_types
    converted_instance = attempt_convert_item(
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/model_utils.py", line 1462, in attempt_convert_item
    raise conversion_exc
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/model_utils.py", line 1453, in attempt_convert_item
    return deserialize_model(input_value, valid_class,
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/model_utils.py", line 1373, in deserialize_model
    return model_class._new_from_openapi_data(**kw_args)
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/model_utils.py", line 45, in wrapped_init
    return fn(_self, *args, **kwargs)
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/model_utils.py", line 369, in _new_from_openapi_data
    return cls._from_openapi_data(*args, **kwargs)
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/model_utils.py", line 45, in wrapped_init
    return fn(_self, *args, **kwargs)
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/model/acl_response.py", line 204, in _from_openapi_data
    composed_info = validate_get_composed_info(
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/model_utils.py", line 1992, in validate_get_composed_info
    allof_instances = get_allof_instances(self, model_args, constant_args)
  File "/opt/homebrew/lib/python3.9/site-packages/fastly/model_utils.py", line 1789, in get_allof_instances
    raise ApiValueError(
fastly.exceptions.ApiValueError: Invalid inputs given to generate an instance of 'ServiceIdAndVersion'. The input data was invalid for the allOf schema 'ServiceIdAndVersion' in the composed schema 'AclResponse'. Error=Invalid type for variable 'version'. Required value type is int and passed type was str at ['received_data'][0]['version']
```

From some debugging I see that it's not changing the type in `remove_uncoercible` function. The type pair `<str>, <int>` is not `UPCONVERSION_TYPE_PAIRS` and `must_convert` is passed False here https://github.com/fastly/fastly-py/blob/ffef2b7bc453660fad99ecfcd87bb7200d177813/fastly/model_utils.py#L1079

**Comment by Integralist** [comment]
Thanks @davidglvn for opening this issue. 

I've created an internal ticket to track this issue. We'll report back once we have an update.

**Comment by Integralist** [comment]
I'm going to close this issue for now as I believe it was resolved in the 2.1.1 release back in May.

https://github.com/fastly/fastly-py/blob/main/CHANGELOG.md
